infrastructure/http/HTTPRequestExecutionAction.go

```diff
@@ -51,32 +50,36 @@ func (instance HTTPAction) Execute(ctx context.Context, executionContext core.Ex
 
 	default:
 
-		if instance.Body != "" && instance.Body[0] == '@' {
-			contents, err := ioutil.ReadFile(instance.Body[1:])
+		var defaults = instance.extractor.Extract(executionContext)
+
+		if instance.State.Body != "" && instance.State.Body[0] == '@' {
+			contents, err := ioutil.ReadFile(instance.State.Body[1:])
 			if err != nil {
 				result[core.ErrorUrn.String()] = err
 				return result
 			}
-			instance.Body = string(contents)
+			instance.State.Body = string(contents)
 		}
 
-		var requestURL = instance.URL
-		var method = instance.Method
-		var headers = http.Header{}
-		var body = instance.Body
+		var requestURL = defaults.URL
+		if instance.State.URL != "" {
```

Comment:
OK so another thought... since these check and assignments are all happening on the `instance.State`, should this not be handled by methods on the `State` object?